File: EqAspNetCoreDemo/Models/Order.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using Korzh.EasyQuery;
namespace EqAspNetCoreDemo.Models
{
    [DisplayColumn("Name")]
    public class Order {

        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int OrderID { get; set; }

        [NotMapped]
        public string Name {
            get {
                return string.Format("{0:0000}-{1:yyyy-MM-dd}", this.OrderID, this.OrderDate);
            }
        }
        
        [Display(Name = "Ordered")]
        public DateTime? OrderDate { get; set; }

        [Display(Name = "Required")]
        public DateTime? RequiredDate { get; set; }

        [Display(Name = "Shipped")]
        public DateTime? ShippedDate { get; set; }

        public decimal? Freight { get; set; }

        public virtual Customer Customer { get; set; }

        public virtual Employee Employee { get; set; }

        public virtual List<OrderDetail> Items { get; set; }

        [ScaffoldColumn(false)]
        public int? ShipVia { get; set; }

        public string ShipName { get; set; }

        public string ShipAddress { get; set; }

        public string ShipCity { get; set; }

        public string ShipRegion { get; set; }

        public string ShipPostalCode { get; set; }

        public string ShipCountry { get; set; }

        public virtual Product Product { get; set; }

        public int EmployeeId { get; internal set; }
    }


}

File: AspNet4/WebForms/VB.AdvancedSearch/Models/NWind/Order.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema


<DisplayColumn("Name")>
Public Class Order
    <DatabaseGenerated(DatabaseGeneratedOption.None)>
    <Column("OrderID")>
    Public Property Id As Integer

    <NotMapped>
    Public ReadOnly Property Name As String
        Get
            Return String.Format("{0:0000}-{1:yyyy-MM-dd}", Me.Id, Me.OrderDate)
        End Get
    End Property

    <Display(Name:="Ordered")>
    Public Property OrderDate As DateTime?
    <Display(Name:="Required")>
    Public Property RequiredDate As DateTime?
    <Display(Name:="Shipped")>
    Public Property ShippedDate As DateTime?
    Public Property Freight As Decimal?
    Public Property CustomerID As String
    <ForeignKey("CustomerID")>
    Public Overridable Property Customer As Customer
    Public Property EmployeeID As Integer?
    <ForeignKey("EmployeeID")>
    Public Overridable Property Employee As Employee
    Public Overridable Property Items As List(Of OrderDetail)
    <ScaffoldColumn(False)>
    Public Property ShipVia As Integer?
    Public Property ShipName As String
    Public Property ShipAddress As String
    Public Property ShipCity As String
    Public Property ShipRegion As String
    Public Property ShipPostalCode As String
    Public Property ShipCountry As String
End Class

